pkg/util/kube.go

```diff
@@ -377,6 +378,31 @@ func GetPodLabelValues(n, label string) (map[string]string, error) {
 	return m, nil
 }
 
+// GetPodAnnotations gets a map annotations from a pod name for the given: namespace, pod label and retry times for checking the pod annotations
+func GetPodAnnotations(n, podName string, retries int) (map[string]string, error) {
+	retry := Retrier{
+		BaseDelay: 1 * time.Second,
+		MaxDelay:  1 * time.Second,
+		Retries:   retries,
+	}
+	var annotations map[string]string
+	_, err := retry.Retry(context.Background(), func(_ context.Context, _ int) error {
+		output, err := Shell("kubectl get pod %s -n %s -o jsonpath='{.metadata.annotations}'", podName, n)
+		if err != nil {
+			Log.Infof("Failed to get pods by label %s in namespace %s: %s", podName, n, err)
```

Comment:
Isn't the error below enough?

Reply:
@fjglira Just curious, after the pod is running, why do we need retry here ? Is there a situation that kubectl get pod -o jsonpath='{.metadata.annotations}' returns unexpected result ?

Reply:
@jewertow I'll make changes here. 
@yxun yes sometimes return a exit code 1, it's failing right now on the test execution sometimes and checking we have two options: make the sleep time more larger( to ensure the pod exist and is ready, I tested in this way and works but is not a clean fix) or make a retry in case that the get annotations fails to ensure we are going to get those annotations